### DISCUSSION TITLE
Inline style on rendered content

### DISCUSSION THREAD

**Discussion by KevinBatdorf** [wp-graphql/wp-graphql#2255] [Ideas]
Gutenberg will enqueue inline styling on posts. If you query the content directly (via WP_GraphQL, for example), these styles get lost.

https://github.com/WordPress/WordPress/blob/master/wp-includes/block-supports/layout.php#L178

One approach to solving this is to just rebuild the content and return the styling with the string, using [`wp_render_layout_support_flag`](https://github.com/WordPress/WordPress/blob/master/wp-includes/block-supports/layout.php#L142)

I created a gist with an example: https://gist.github.com/KevinBatdorf/daec9345115279f1c9fe49deb589882f

There may be other (better) approaches, but one way is to modify the function to return the styling with the content. One caveat is that the `render_block` filter is run after other filters have been applied, so you may miss out on something if you're using other pre render filters.

Another approach would be to use `wp_get_layout_style` and parse the block first for the container name and layout. I haven't attempted this as the above works for me so far. I'll update if any new developments appear.

**Comment by ayohdev** [comment]
> Gutenberg will enqueue inline styling on posts. If you query the content directly (via WP_GraphQL, for example), these styles get lost.
> 
> https://github.com/WordPress/WordPress/blob/master/wp-includes/block-supports/layout.php#L178
> 
> One approach to solving this is to just rebuild the content and return the styling with the string, using [`wp_render_layout_support_flag`](https://github.com/WordPress/WordPress/blob/master/wp-includes/block-supports/layout.php#L142)
> 
> I created a gist with an example: https://gist.github.com/KevinBatdorf/daec9345115279f1c9fe49deb589882f
> 
> There may be other (better) approaches, but one way is to modify the function to return the styling with the content. One caveat is that the `render_block` filter is run after other filters have been applied, so you may miss out on something if you're using other pre render filters.
> 
> Another approach would be to use `wp_get_layout_style` and parse the block first for the container name and layout. I haven't attempted this as the above works for me so far. I'll update if any new developments appear.

Thank you so much! this has helped solve a huge problem we have been trying to solve.

**Reply by quartopianodesign** [reply]
Hi, 
updating wp core to 6.2.2 doesn't work anymore, any suggestions?

**Reply by KevinBatdorf** [reply]
Hey, sorry. Just saw this. I commented on [the linked gist](https://gist.github.com/KevinBatdorf/daec9345115279f1c9fe49deb589882f?permalink_comment_id=4623840#gistcomment-4623840) but will leave the comment here too.

It works still for me (as far as I can tell!) but another option would be to use `pre_render_block` and load them in yourself.

```php
add_filter('pre_render_block', function($pre_render, $parsed_block) {
    // $pre_render is null
   
    // Get the block id somehow to use when registering the script below (peek into $parsed_block)
    $id = 123;

    // Get the styles, possibly using the method in the gist above
    $css = 'body { color: red; }';

    wp_register_style("block-{$id}", false, [], null);
    wp_enqueue_style("block-{$id}");

    // Uses wp_strip_all_tags - WP approved approach to safe echoing inline styles
    // See: https://github.com/WordPress/WordPress/blob/9ecfdd8e5a42ed4b66ffecb88321242717c89831/wp-includes/theme.php#L1920
    wp_add_inline_style("block-{$id}", wp_strip_all_tags($css));

    return $pre_render;
}, 10, 2);
```

**Reply by KevinBatdorf** [reply]
Actually, that's not right. I forgot about the purpose of this ticket :D The code above actually is for inlining new css into non-headless WP, so the exact inverse. Oops. Well, it's a good snippet to bookmark anyway. `render_block` should work still, but if you share some code broken I can try running it.
